### PR TITLE
[libc++][format] Fixes formatting code units as integers.

### PR DESCRIPTION
**Pull request by mordante**
This paper was voted in as a DR, so it's retroactively enabled back to C++20; the C++ version that introduced std::format.

Implements:
- P2909R4 Fix formatting of code units as integers (Dude, where’s my ``char``?)

### CODE REVIEW

**Review by ldionne — CHANGES_REQUESTED** [review]

File: libcxx/test/std/utilities/format/format.formatter/format.formatter.spec/formatter.char.fsigned-char.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  }
+#  ifndef TEST_HAS_NO_WIDE_CHARACTERS
+  else {
+    test(STR(R"('\u{80}')"), STR("?}"), '\x80');
+    test(STR("'\u00ff'"), STR("?}"), '\xff');
+  }
+#  endif // TEST_HAS_NO_WIDE_CHARACTERS
+#endif   // TEST_STD_VER > 20
+
+  test(STR("10000000"), STR("b}"), char(-128));
```
Comment (mordante) [inline libcxx/test/std/utilities/format/format.formatter/format.formatter.spec/formatter.char.fsigned-char.pass.cpp:87]: @ldionne These values are different, for the unsigned test I use non-negative values.

File: libcxx/test/std/utilities/format/format.formatter/format.formatter.spec/formatter.char.fsigned-char.pass.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
+//===----------------------------------------------------------------------===//
```
[the remark this answers is not in the document]
Comment (mordante) [reply libcxx/test/std/utilities/format/format.formatter/format.formatter.spec/formatter.char.fsigned-char.pass.cpp:1]: They are not the same. See my other comment where they differ.

**Review by mordante — COMMENTED** [review]
Thanks for the review!

**Review by ldionne — APPROVED** [review]
LGTM -- please apply the clang-format diff to appease it though.

**Comment by mordante** [comment]
> LGTM -- please apply the clang-format diff to appease it though.

Thanks for the review. It seems I locally have an older clang-format in my script.

**Pull request merged**